### ISSUE TITLE
Feature: Add Environment Variables for AppData

### ISSUE THREAD

**Issue by Meberem** [microsoft/azure-pipelines-agent#3056]
TLDR: Would it be possible to add environment variables corresponding to the [SpecialFolder](https://docs.microsoft.com/en-us/dotnet/api/system.environment.specialfolder?view=netcore-3.1) enum values?

I am trying to cache some files as part of my build. Fortunately, the [tool in question](https://github.com/aspnet/LibraryManager/blob/c94888576f2387959832db6c9f38bcb8becd8596/src/LibraryManager/CacheService/CacheService.cs#L47) uses a cache. The cache is located somewhere under the `Environment.SpecialFolder.LocalApplicationData` folder. After a bit of research, I can see that it is `/home/vsts/.local/share/` happy days. It would be amazing if there was a variable like `$(SpecialFolder.LocalApplicationData)` that would resolve to it.

**Comment by Maxwell175** [comment]
This feature would also be useful to us. Many utilities hold their caches in AppData or something like ~/.cache on Linux.
